### PR TITLE
Update mechset reading and icon tests

### PR DESCRIPTION
**Pull request by SJuliez**
Updates some icon tests (mechset, image files, using chassis/generic icons) as well as the read code for mechset itself. Not safe to pull for .19.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/tileset/MechTileset.java [hunk 1]
```diff
@@ -353,46 +359,32 @@
...
-                    && st.sval.equalsIgnoreCase("include")) {
-                st.nextToken();
-                name = st.sval;
-                LogManager.getLogger().debug("Loading more unit images from " + name + "...");
-                try {
-                    loadFromFile(name);
-                    LogManager.getLogger().debug("... finished " + name + ".");
-                } catch (IOException e) {
-                    LogManager.getLogger().debug("... failed: " + e.getMessage() + ".", e);
+        LogManager.getLogger().info("Loading unit icons from {}", filename);
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tileset/MechTileset.java:362]: doesnt include the directory

File: megamek/src/megamek/client/ui/swing/tileset/MechTileset.java [hunk 2]
```diff
@@ -353,46 +359,32 @@
...
+                            } catch (IOException e) {
+                                LogManager.getLogger().error("... failed: {}.", e.getMessage(), e);
+                            }
+                        } else if (tokens.get(0).equals(CHASSIS_KEY)) {
+                            chassis.put(tokens.get(1).toUpperCase(), new MechEntry(tokens.get(2)));
+                        } else {
+                            exact.put(tokens.get(1).toUpperCase(), new MechEntry(tokens.get(2)));
+                        }
+                    } else {
+                        LogManager.getLogger().warn("Malformed line in {}: {}", filename, tokens.toString());
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tileset/MechTileset.java]: doesnt include the directory

File: megamek/src/megamek/client/ui/swing/tileset/MechTileset.java [hunk 3]
```diff
@@ -353,46 +359,30 @@
...
+                        } catch (IOException e) {
+                            LogManager.getLogger().error("... failed: {}.", e.getMessage(), e);
+                        }
+                    } else if (tokens.get(0).equals(CHASSIS_KEY)) {
+                        chassis.put(tokens.get(1).toUpperCase(), new MechEntry(tokens.get(2)));
+                    } else {
+                        exact.put(tokens.get(1).toUpperCase(), new MechEntry(tokens.get(2)));
+                    }
+                } else {
+                    LogManager.getLogger().warn("Malformed line in {}: {}", filename, tokens.toString());
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tileset/MechTileset.java:382]: doesnt include the directory

File: megamek/src/megamek/common/util/fileUtils/StandardTextfileStreamTokenizer.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public StandardTextfileStreamTokenizer(Reader r) {
+        super(r);
+        eolIsSignificant(true);
+        commentChar('#');
+        quoteChar(DOUBLE_QUOTE);
+        wordChars('_', '_');
+    }
+
+    public StandardTextfileStreamTokenizer(InputStream is) {
+        super(is);
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/common/util/fileUtils/StandardTextfileStreamTokenizer.java:59]: We use InputStream widely

**Review by HammerGS — APPROVED** [review]
Tested and seems to be working fine.

**Pull request merged**